README.md

```diff
@@ -1,22 +1,381 @@
+If your function is supposed to not return anything you can omit value in return statement.
+
+
+```javascript
+const square = fun(x) {
+    return x * x;
+};
+
+square(4); // returns 16
+
+
+const printSquare = fun(x) {
+    print(x * x);
+    
+    return;
+};
+
+printSquare(4); // prints 16, returns null.
+```
+
+###### Arrays
+
+`[` `expressions...` `]`
+
+Arrays in Junior are as immutable as any other literals.
+They are not bound to one type but can store values of different types.
+Arrays are indexed starting from 0.
+
+```javascript
+const arr = [true, 2, "three", fun(x) { return x * x; }];
+
+arr[3](6); // 36
+```
+
+###### Hashes
+
+`{` `primitive type literal` `:` `expression` ... `}`
+
+Hashes are maps with key - value pairs.
+They are similar to Javascript's objects. Check them out:
+
+```javascript
+const obj = { "name": "John Doe", 4: "Been there.", "greet": fun(name) { return "Hi " + name + "! I'm John Doe"; } };
+
+const greetStr = "greet";
+
+print(obj[4]); // Been there.
+print(obj[greetStr]("Jane")); // Hi Jane! I'm John Doe 
+```
+
+##### Operations
+
+Junior supports many operations, from adding to numbers to retrieving value from a hash or array.
+Here is a list of Junior's operations in order of their precedence.
+
+
+###### Logical 
+
+operators: `==`, `!=`, `>=`, `<=`, `>`, `<`
+
+They evaluate and return logical value of expression they represent.
+> Note that as for now they also support primitive types (booleans, integers, strings) as their operands.
+
+###### Mathematical:
+
+operators: `+`,`-`, `*`, `/`
+
+Those operators return result of mathematical operation evaluated between their operands.
+They only support integers as their operands.
+
+```javascript
+30 + 12;
+84 / 2;
+1 * 42;
+42 - 0;
+```
+
+###### Concatenation
+
+operator: `+`
+
+Adds together two strings and returns the result.
+
+```javascript
+"Hello" + " World!";
+```
+
+###### Number Negation
+
+operator: `-`
+
+Prefixed operator for negating an integer.
+
+```javascript
+52 + -10;
+```
+
+###### Boolean Negation
+
+operator: `!`
+
+Prefixed operator for negating a boolean expression.
+
+```javascript
+const truth = true;
+!truth;
+```
+
+###### Function Call
+
+operators: `()`
+
+To call a function put parenthesis after identifier or any other expression that evaluates to a function literal and pass arguments between them.
+
+```javascript
+const myWeirdFunction = fun(x) {
+    return x + 2;
+};
+
+myWeirdFunction(40);
+
+fun() {
+    print("This is an IIFE!");
+}();
+```
+
+###### Retrieving value with Index
+
+operators: `[]`
+
+The bracket operators are used to retrieve values from arrays and hashes.
+They work just as in any other language.
+
+```javascript
+const myArray = [4, 2, 0];
+
+myArray[0]; // 4
+
+const theUniverse = { 42: "the answer", "isEarthFlat": false };
+
+theUniverse[42];
+theUniverse["isEarthFlat"];
+```
+
+##### Identifiers
+
+Identifiers are also treated as expressions.
+They evaluate to expression they are bind to.
```

Comment:
they are bind to should be changed but not sure what is the correct form.. xd
bound to?

Reply:
yup